[Repository Management] Remove reposman functionality.

oliverguenther opened the pull request:
This commit removes the functionality provided by reposman.rb,
and replaces that script with an empty placeholder printing only
a deprecation warning (as to not throw errors in existing cronjobs).

In detail, this commit does:
- replace reposman.rb with an script printing a deprecation only.
- Remove repositories creation API in SysController and corresponding
  specs
- Update documentation guides concerning the deprecation of reposman
- Remove packager scripts for creating a reposman.rb cronjob.
## TODOS
- [x] Add rake tasks to migrate previously created _quasi-managed_ repositories created by reposman to repositories managed by OpenProject.

Relevant work package: https://community.openproject.org/work_packages/21029

oliverguenther committed:
[ci skip] Remove reposman functionality.

This commit removes the functionality provided by reposman.rb,
and replaces that script with an empty placeholder printing only
a deprecation warning (as to not throw errors in existing cronjobs).

In detail, this commit does:

* replace reposman.rb with an script printing a deprecation only.
* Remove repositories creation API in SysController and corresponding
specs
* Update documentation guides concerning the deprecation of reposman
* Remove packager scripts for creating a reposman.rb cronjob.
* Adds a rake task to migrate repositories to managed type with a common
* URL prefix.

Commit:
intitial display of error messages via the notification service

Signed-off-by: Florian Kraft <f.kraft@finn.de>

Commit:
sneak in a fix for the attachments

this will corect an error thrown when a validaiton error happens -
the form will, in certain edge cases try to submit the work package files
which can be empty at times.

Signed-off-by: Florian Kraft <f.kraft@finn.de>

Commit:
show the errors when the requests have completed

Signed-off-by: Florian Kraft <f.kraft@finn.de>

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://oliverguenther-feature-deprecate-reposman.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://oliverguenther-feature-deprecate-reposman.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

Commit:
actually use the notifciations in protractor tests

Signed-off-by: Florian Kraft <f.kraft@finn.de>

Commit:
correct test for new error notification

Signed-off-by: Florian Kraft <f.kraft@finn.de>

Commit:
remove obsolete error tests

Signed-off-by: Florian Kraft <f.kraft@finn.de>

ulferts commented on packaging/scripts/create-svn-repositories:
```diff
@@ -1,24 +0,0 @@
-  --url "file://${SVN_REPOSITORIES}" \
-  --key "${SYS_API_KEY}" \
-  --scm Subversion \
-  --verbose
```
@kgalli, @crohr 

There are some changes in the management of repositories. In short, reposman.rb is no longer required. Instead, repository creation is handled directly from within OpenProject.

This will also need to be reflected by the packaging. The adaptations required should encase:
- Configuring the repository options, please see:
  - https://community.openproject.org/work_packages/21050
  - https://github.com/opf/openproject/blob/release/4.3/config/configuration.yml.example#L227 
- If possible, remove the remains of the cron job configured on existing installations.
- If possible, call a rake task on existing installations to bring already configured installations under OpenProject management (Not of the utmost importance).

ulferts commented on doc/subversion_and_git_integration.md:
```diff
@@ -94,23 +94,32 @@ PerlLoadModule Apache::OpenProjectAuthentication
+To convert repositories previously created by reposman.rb into managed repositories,
+use the following command:
+
+    $ bundle exec rake scm:migrate:reposman[URL prefix (, URL prefix, ...)]
```
I don't think this should be limited to repos having been created via reposman. In a scenario, where a repository is moved to openproject for one reason or the other, this functionality is necessary as well.

oliverguenther replied:
I agree, I've updated the rake naming and description to be independent from reposman.

ulferts commented:
Looks good, but I will wait for @crohr, @kgalli or @jonasheinrich to make a statement regarding the necessary changes to the packaged installation.

kgalli commented:
@ulferts, @oliverguenther we will have to adjust that packager `configuration.yml` file. We will also have to rework the [svn-addon](https://github.com/pkgr/addon-svn-dav). Maybe we can skip the addon at all. Right now I'm not sure about the whole [vhost](https://github.com/pkgr/addon-svn-dav/blob/master/conf/vhost/svn_dav.conf) config though.

oliverguenther committed:
[ci skip] Remove specific reference to reposman from the migrate rake task

This commit alters the rake task to make it verbally independent
from the reposman functionality.

oliverguenther commented:
@kgalli As the introduced change for 4.3 does not impact authorization, the `svn-addon` remains relevant and does not need to be updated, as the authorization routes of the sys API remain the same.

kgalli commented:
@oliverguenther I was more concerned about the naming and the `svn_dav`. Maybe we should change the add on name to `repository` and have a look what we have to change in regard to the Apache config? What do you think?

oliverguenther commented:
Ah, I see what you mean.

This change will become relevant in the context of https://github.com/opf/openproject/pull/1629, when the Apache module may serve Git repositories as well.

I can foresee two options:
1. A consolidated addon for svn and git both (repository would be a suiting name)
2. A separate addon for git, as it will probably be superfluous in some of our installations.

kgalli commented:
@oliverguenther I think we can do it both ways without the superfluous in some installations. We just have to ensure that the user choose the repo-configuration he wants to be installed. So if he wants svn and no git he can do that. Regardless of if we decide to use two add-ons or a combined one. So which approach do you like the most. Consolidated add-on or two separate ones?

oliverguenther commented:
While I would usually prefer a single addon for both for reduced effort, in this case I would prefer to be able to physically remove the whole (Git, SVN) addon from Apache should it not be chosen by the installer. Otherwise we would have to disable a part of the provided addon.

ulferts committed:
prevent admins from moving a wp into unfitting project

Unfitting means, that the work package module is not enabled in the
project. There would have been an alternative implementation but it is
probable, that the implementation choosen can be generalized

ulferts committed:
use existing method for permission check

ulferts committed:
explicitly hand over the current user

oliverguenther committed:
Remove superfluous route

ulferts committed:
prevent moving to not allowed project in work package model

ulferts committed:
extract work package move/copy to own service

ulferts committed:
merge specs for copy and move

ulferts committed:
consolidate specs for move work package service

ulferts committed:
remove instances of User.current

ulferts committed:
scoping specs

ulferts committed:
extract method

ulferts committed:
please rubocop

Commit:
replace changed_data by details

working towards making the alias obsolete...

ulferts committed:
extract methods to reduce complexity

ulferts committed:
further method extraction

ulferts committed:
adapt spec to changed interface

crohr commented:
Regarding packages, I think the best way would be to auto-update any legacy repos in the [openproject addon postinstall](https://github.com/pkgr/addon-openproject/blob/master/bin/postinstall). Am I right that the necessary configuration option can be set as an ENV variable? If so the addon could detect whether it is already set or not, and if not it would call the rake task to migrate, and then set the variable so that it is not performed in subsequent updates. 

Regarding the svn addon, from what I understand it does not need to be changed since you still need the authorisation to be performed when accessing /svn via http.

oliverguenther commented:
@crohr You're right that the svn-addon itself does not need to be changed, however there's a pending pull requests that integrates the same authorization functionality for Git (Smart HTTP) into the Apache module and /sys API here: https://github.com/opf/openproject/pull/3289

With that change in mind, I've extended the svn-addon into a single addon that configures Apache for both SVN and Git. https://github.com/finnlabs/addon-repositories

Concerning legacy migration of existing repositories served over Apache, we had the same idea in mind. [This rake task](https://github.com/oliverguenther/openproject/blob/feature/deprecate_reposman/lib/tasks/scm.rake) will be added to the core to migrate a given path prefix to managed repositories.

In the postinstall, I take it we can access the SVN directory env variable as the prefix to these repositories.

---

A remaining question regarding configuration of SVN and Git, how would you allow the user to disable one (or both) so that the Apache configuration is not written? Can you briefly walk me through how this is done for svn-addon?

(I would believe that this is something the user should be allowed to choose in the wizard, e.g., through a multi checkbox option - but I'm unsure what capabilities the wizard provides at the moment).

ulferts committed:
sort version by name

ulferts committed:
consolidate duplicate code

ulferts committed:
separating viewmodel and view logic

NobodysNightmare committed:
display diff of containing aggregated journal

... instead of displaying the diff for the requested journal only

- this should provide the best forward compatibility of links
- even when aggregation timeframes are changed intermittently

ulferts committed:
initialize files to prevent operations on undefined

oliverguenther committed:
Merge pull request #3302 from ulferts/fix/initialize_empty_files_array

initialize files to prevent operations on undefined

NobodysNightmare committed:
Merge branch 'release/4.3' into dev

NobodysNightmare committed:
filter for right journal on SQL level

NobodysNightmare committed:
bump bundler to 1.10.6

oliverguenther committed:
Merge pull request #3303 from NobodysNightmare/feature/bump_bundler_version

bump bundler to 1.10.6

NobodysNightmare committed:
show comment in WP created mail

HDinger committed:
added style rule to resize the font in paragraph, so that all fields have the same size

HDinger committed:
Merge branch 'dev' into fix/20740-text-size-CF

NobodysNightmare committed:
show initial journal in activity tab

NobodysNightmare committed:
add spec for hiding details on first journal

oliverguenther committed:
Merge branch 'release/4.3' into dev

ulferts committed:
unit testing inplace editor dropdown directive

NobodysNightmare committed:
ensure aggregation is disabled when testing diff

alternative would be to ensure that the change is done "late" enough or by a different user.
I hope this way is the most explicit one.

ulferts committed:
remove duplicate assignment of custom field values

The assignments happens in work_package.copy_from as well

ulferts committed:
remove superfluous safeguard

HDinger committed:
Changed the line height so that the lines don't overleap and everything is readable

HDinger committed:
changed the nesting so that angular-generated classes where avoided

NobodysNightmare committed:
hide inactive priorities when applicable

NobodysNightmare committed:
fix specs

HDinger committed:
made the line height a bit smaller

ulferts committed:
sort versions throughout the rails world

ulferts committed:
Merge pull request #3310 from NobodysNightmare/feature/hide_inactive_priorities

hide inactive priorities when applicable

NobodysNightmare committed:
Merge pull request #3309 from HDinger/fix/20629-overleaping-lines-in-diff

[20629] overleaping lines in diff

HDinger committed:
the last CF with long text doesn't the large marge any more

NobodysNightmare committed:
Merge pull request #3297 from ulferts/fix/filter_allowed_projects_on_move

Fix/filter allowed projects on move

HDinger committed:
fixed undesired line break in search

HDinger committed:
made the search result site consistent with trailing-labels

ulferts committed:
fix attachments download link when on sub uri

Using the url_helpers without specifying the :host option generates urls
not scoped to the sub uri. The OpenProject::StaticRouting::UrlHelpers
provides the required functionality

ulferts committed:
remove unnecessary extend work around

HDinger committed:
Changed the position of enumerations so that they are all on one line

ulferts committed:
ensure sub uri fitness of all api paths

All specs now use a share example which will also test for sub uri's.
Additionally, the tests are more focused now as duplication is avoided.

ulferts committed:
Merge pull request #3312 from HDinger/fix/21145-line-breaks-in-search

[21145] line breaks in search

Commit:
Bump version to 5.0.0-alpha

ulferts committed:
remove superfluous shared examples

NobodysNightmare committed:
enforce disabled aggregations

due to limited time resolution we can't assume that a setting of 0 will implicitly disable aggregation

NobodysNightmare committed:
correctly render initial activity in overview

ulferts committed:
reduce duplication in directive test

oliverguenther committed:
Remove repository project hierarchy

This commit removes the project hierarchy introduced with
repository management.
They are not required, incompatible with mod_dav_svn and provide
little to none advantage to storing all repositories in a flat
structure for the scale we require.

HDinger committed:
changed the margin-botton to an absout value instead of heritage

NobodysNightmare committed:
Merge pull request #3304 from HDinger/fix/20740-text-size-CF

[20740] text size of cf

HDinger committed:
Removed unnecessary code, which became obsolete because of the previous changes

NobodysNightmare committed:
fix added menu items after query save as

- links would be wrong on subdirectory installations

oliverguenther committed:
Disallow setting the owner of a repository folder

The original repository management epic allows to define
an owner that a created repository is set to.

As this will not work for non-root users on *nix systems,
this commit removes this functionality in favor of setting
only the group.

This is allowed by non-privileged users _as long as_ users are
member of this group.

oliverguenther committed:
Repository form: Toggle not only inputs, but selects

When toggling the SCM type, form elements must be disabled manually
due to a big with fieldset[disabled] on IE10/11.

This commit adds the missing select selector to catch the
path_encoding select re-introduced in https://github.com/opf/openproject/pull/3296

NobodysNightmare committed:
fix link to version

NobodysNightmare committed:
Merge pull request #3315 from ulferts/fix/attachment_download_link

Fix/attachment download link

oliverguenther committed:
Merge pull request #3301 from NobodysNightmare/feature/aggregated_diffs

Aggregate diffs

oliverguenther committed:
Merge pull request #3318 from NobodysNightmare/feature/fix_query_save_as

fix links in subfolder deployments

NobodysNightmare committed:
let delayed jobs reset the request store

This is especially neccessary since our Settings use the request store to determine, whether they need to invalidate the cache key.
If the request store was never reset, the settings in our delayed jobs would be cached at an old value forever (until the next restart).

oliverguenther committed:
Fix configuration.yml important notice position

NobodysNightmare committed:
Merge pull request #3319 from oliverguenther/fix/repository_ownership

Disallow setting the owner of a repository folder

NobodysNightmare committed:
Merge pull request #3320 from oliverguenther/fix/repository_disable_select

Repository form: Toggle not only inputs, but selects

NobodysNightmare committed:
Merge pull request #3314 from HDinger/fix/21187-consistent-enumeration

[21187] consistent enumerations

ulferts committed:
provide the configuration upon compile

This hopes to fix a test failure only occuring on travis.

ulferts committed:
Merge pull request #3307 from NobodysNightmare/feature/show_initial_comments

Show comments of initial journal

ulferts committed:
Merge branch 'release/4.3' into dev

HDinger committed:
started fixing the syling of the new table component

oliverguenther committed:
[21047] Work package activities also display repository revisons

This commit adds preliminary support for showing aggregated regular work
package activities and revisions linked to that particular work package
(through commit message referencing).

Related work packages:
https://community.openproject.org/work_packages/15422
https://community.openproject.org/work_packages/21047

ulferts committed:
disabling test failing on travis

NobodysNightmare committed:
generalize module included in all delayed jobs

now also applies to other activities one might want to execute before all delayed jobs
(as discussed with @ulferts)

NobodysNightmare committed:
Merge pull request #3317 from oliverguenther/feature/managed_repos_structure

Remove repository project hierarchy

HDinger committed:
styled the new table component according to the working package table

oliverguenther committed:
Specify changes to APIv3 revisions endpoint

This commit introduces two changes to the revision endpoint:

1. A `formattedIdentifier` which may either contain a shortened
identifier (e.g., an unambiguous SHA-1 hash fraction for Git) or be identitical to
`identifier` (for SVN).

2. A `showRevision` link which escapes the scope of the APIv3, but
provides a reference to the repository view on that particular revision
entry.

[ci skip]

NobodysNightmare committed:
hide global queries from project-scoped WP lists

myabc committed:
Merge branch 'dev' into feature/rails4

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Temp. switch openproject-translations to PR branch

Revert once Pull Request merged:
https://github.com/opf/openproject-translations/pull/27

Signed-off-by: Alex Coles <alex@alexbcoles.com>

NobodysNightmare committed:
Merge pull request #3326 from oliverguenther/feature/repository_endpoint_formatted_identifier

[APIv3] Specify changes to revisions endpoint

HDinger committed:
copied the sass from wp table for the generic table

oliverguenther committed:
Extend revision representer with identifier and revision link

This commit extends the revision endpoint with a link
to the (non-api) revision view as well as the formatted
identifier from the Changeset model.

NobodysNightmare committed:
specify WP list endpoint

NobodysNightmare committed:
move doc for same endpoint together

- all /api/v3/projects/{id}/work_packages now reside under "work packages"
- slightly unified writing of "Work Package" (at least in titles)

NobodysNightmare committed:
specify auxiliary links to work package list

[ci skip]

NobodysNightmare committed:
Merge pull request #3308 from ulferts/fix/sort_version_by_name

Fix/sort version by name

oliverguenther committed:
Add missing path spec

NobodysNightmare committed:
make collection representer explicitly unpaginated

HDinger committed:
the work packages table is now inheriting from the generic table

HDinger committed:
rebuild the correct nesting

HDinger committed:
removed unnecessary variables

oliverguenther committed:
Use ng-switch to determine activity type

Commit:
split up the individual parts of the documentation for the frontend

Signed-off-by: Florian Kraft <f.kraft@finn.de>

oliverguenther committed:
Move activities out of tabs

This moves activity templates and directives in their own
namespaces / module.
Also adds a single activity entry for overview and activity,
which itself selects the correct directive for the given activity
object.

oliverguenther committed:
Remove superfluous path

Revision path is now returned from API endpoint and thus
can be removed from the path helper.

oliverguenther committed:
Add missing href to resource

HDinger committed:
corrected naming so that colors menu item gets correct background when selected

HDinger committed:
made the side bar pixel accurate

NobodysNightmare committed:
first version of a paginated collection

- common base class with UnpaginatedCollection
- one implementor

NobodysNightmare committed:
split specs for unpaginated and paginated collections

ulferts committed:
enable directives in uiComponents to be used in styleguide

ulferts committed:
remove col tags in dom

Those where left overs from the time where we always had the ID column and there was the "i" to trigger the details pane

ulferts committed:
extraxt column highlighting into own directive

The directive is usable for every table and is used right away by the work packages table.

HDinger committed:
words in the field label are broken now if they are so long that they overlap the field

NobodysNightmare committed:
actually implement endpoint for WPs by projects

Commit:
minor overview of the folder structure for frontend

NobodysNightmare committed:
Merge pull request #3330 from HDinger/fix/20900-blue-background-missing-colors-menu-item

[20900] Blue background missing for 'Colors' menu item

Commit:
add a document with details on the build process

Signed-off-by: Florian Kraft <f.kraft@finn.de>

HDinger committed:
first effort to limit the text when long words are contained

oliverguenther committed:
Add unit test for revision activity

NobodysNightmare committed:
Merge pull request #3328 from oliverguenther/feature/revision_endpoint_changes

Extend revision representer with identifier and revision link

Commit:
initial version for testing

Signed-off-by: Florian Kraft <f.kraft@finn.de>

Commit:
make a proper list

Signed-off-by: Florian Kraft <f.kraft@finn.de>

NobodysNightmare committed:
Merge pull request #3332 from HDinger/fix/21142-field-name-overlaps-field

[21142] Field name overlaps field box in modal view

Commit:
Removes unused strings

oliverguenther committed:
Merge pull request #3335 from jonasheinrich/fix/delete_unused_strings

Removes unused strings

NobodysNightmare committed:
Merge pull request #3331 from HDinger/fix/20788-project-menu-pixel-accurate

[20788] project menu is not pixel accurate

HDinger committed:
fixed bug in toggled sidebar

oliverguenther committed:
Merge pull request #3325 from NobodysNightmare/feature/exclude_global_queries_from_project

hide global queries from project-scoped WP lists

NobodysNightmare committed:
Merge pull request #3333 from ulferts/feature/highlight-column-directive

Feature/highlight column directive

ulferts committed:
Merge pull request #3321 from NobodysNightmare/feature/reset_request_store_in_jobs

Reset RequestStore before running delayed jobs

basiszwo committed:
Merge pull request #3329 from NobodysNightmare/feature/specify_wp_index

Specification for WP list endpoint

basiszwo committed:
Merge pull request #3334 from NobodysNightmare/feature/api_wp_list

[API] Implementation of WP list endpoint

Commit:
Update OpenProject-Translations

Commit:
add API topics

Signed-off-by: Florian Kraft <f.kraft@finn.de>

HDinger committed:
made sure not only the title is truncated but other fields in details pane too

HDinger committed:
Merge branch 'dev' into fix/20908-new-table-component

HDinger committed:
inserted the classes responsible for highlighting the colums when hovered

ulferts committed:
Merge pull request #3322 from HDinger/fix/20908-new-table-component

[20908] new table component

Commit:
add a note on mocking stuff

Signed-off-by: Florian Kraft <f.kraft@finn.de>

Commit:
describe the plugins and their tests and include notes on commands

Signed-off-by: Florian Kraft <f.kraft@finn.de>

ulferts committed:
fix my page styling in IE

Middle section should be equally 50% left and 50% right

ulferts committed:
Merge pull request #3340 from ulferts/fix/my_page_left_right_layout

fix my page styling in IE

Commit:
add a note on duplication involving the hal-api-resource

[ci skip]

Signed-off-by: Florian Kraft <f.kraft@finn.de>

HDinger committed:
used new tabe component design for projects table

oliverguenther committed:
Merge branch 'release/4.2' into dev

ulferts committed:
Merge pull request #3337 from opf/fix/update_translations

Update OpenProject-Translations

HDinger committed:
if buttons are inside a table, they will be aligned on the right side

Commit:
add notes on the styling of OP

Signed-off-by: Florian Kraft <f.kraft@finn.de>

HDinger committed:
fixed little design bugs within in the table

Commit:
add notes on how to use partials in the styleguide

[ci skip]

Signed-off-by: Florian Kraft <f.kraft@finn.de>

ulferts committed:
prevent spec flicker by cleaning up after Setting.notified_events

Commit:
add examples for using index.js

[ci skip]

Signed-off-by: Florian Kraft <f.kraft@finn.de>

ulferts committed:
adapt protractor spec to avoid flicker

ulferts committed:
finally disabling unreliable protractor spec

ulferts committed:
create interactive table directive

This is mostly c&p from the work packages table for now

NobodysNightmare committed:
Merge pull request #3336 from HDinger/fix/20788-project-menu-pixel-accurate

[20788] project menu is not pixel accurate (2)

ulferts committed:
apply interactive-table to work-packages-table

Commit:
add notes on coverage and dummy services in testing

Signed-off-by: Florian Kraft <f.kraft@finn.de>

Commit:
add hints for template handling

[ci skip]

Signed-off-by: Florian Kraft <f.kraft@finn.de>

ulferts committed:
generalize work package table dom even more

The only thing that remains to be specific is the
"work-package-table--container" class as the beginning of the element
block. With it, it should be possible to later on apply specific styles
to the work packages table again

oliverguenther committed:
Have tests use new work package table rules

ulferts committed:
Merge pull request #3 from oliverguenther/fix/interactive_table_tests

Have tests use new work package table rules

oliverguenther committed:
Fix support css rule

ulferts committed:
fix selector

HDinger committed:
menu item gets correct background when being selected

oliverguenther committed:
Allow checkout plugin to display information below empty notification

This calls a hook in the repositories/empty template to allow
the checkout plugin to display checkout information there.

[ci skip]

NobodysNightmare committed:
Merge pull request #3349 from HDinger/fix/21239-background-missing-time-sheet

[21239] background missing for menu item 'Time sheet'

NobodysNightmare committed:
Merge pull request #3351 from oliverguenther/fix/empty_repository_hook

Allow checkout plugin to display information below empty notification

ulferts committed:
Merge pull request #4 from oliverguenther/fix/interactive_table_tests

Fix support css rule

oliverguenther committed:
Fix broken cuke

HDinger committed:
Merge pull request #3 from oliverguenther/fix/restyle-projects-table-cuke

Fix broken cuke

oliverguenther committed:
Merge pull request #3346 from ulferts/feature/interactive_table_directive

Feature/interactive table directive

oliverguenther committed:
Merge pull request #3342 from HDinger/feature/18800-restyle-projects-table

[18800] Restyle and restructure projects table in admin overview

oliverguenther committed:
Correct for wrong isInitial

oliverguenther committed:
Revision integration tests

oliverguenther committed:
Merge branch 'release/4.2' into dev

floriank committed:
typo

oliverguenther committed:
Add better contextual view hooks to navigation and empty view

This commit adds distinguishable view hooks for plugins wanting
to:

* Add a toolbar item to the repository navigation
* Show contextual information specifically on the empty repository
  screen

Commit:
add additional hints about working with the legacy code base

[ci skip]

Signed-off-by: Florian Kraft <f.kraft@finn.de>

ulferts committed:
using stable expectation

Commit:
add some notes on the legacy frontend

[ci skip]

Signed-off-by: Florian Kraft <f.kraft@finn.de>

Commit:
typo hunting an phrasing

[ci skip]

Signed-off-by: Florian Kraft <f.kraft@finn.de>

ulferts committed:
hoping for test stability

floriank committed:
remove a sentence implying no ticket endpoint

This was false information, as the ticket endpoint has been added a while ago.

NobodysNightmare committed:
Merge pull request #3341 from floriank/feature/frontend-documentation

Frontend documentation

ulferts committed:
save project before mocking .all method

ulferts committed:
remove expectation on mocked method

Commit:
only show errors when there are errors to be shown

Errors were alwas shown by accident. As the EditableFieldsState object is not
cleared upon successful request, using `finally` would show the error once more
for a successful api call.

Signed-off-by: Florian Kraft <f.kraft@finn.de>

oliverguenther committed:
Fix invalid label concerning storage overview

oliverguenther committed:
Merge pull request #3360 from oliverguenther/fix/storage_invalid_i18n

Fix invalid label concerning storage overview

ulferts committed:
update database cleaner to work around failing spec

A transaction was used in the production code the spec covered. The issue is described here: https://github.com/DatabaseCleaner/database_cleaner/issues/234

basiszwo committed:
Strip all passed values for non-array values

basiszwo committed:
Add rake task to cleanup merged (remote) branches

[ci skip]

timhabermaas committed:
[Refactor] Use method memoization instead of explicitly setting ivar.

timhabermaas committed:
[Refactor] Simplify User.register_allowance_evaluator.

timhabermaas committed:
[Refactor] Use ActiveRecord#find_by_id instead of #find and rescue.

timhabermaas committed:
[Refactor] Make wp_hashes_with_string depend solely on its parameters.

timhabermaas committed:
[Refactor] Remove useless setters; remove useless mutation of self.

ulferts committed:
change database_cleaner strategy

Due to https://github.com/DatabaseCleaner/database_cleaner/issues/345

ulferts committed:
adapt specs to message_chain

oliverguenther committed:
Move project required disk space counting to sql

This adds a model section `Project::Storage` that allows
counting required disk space for wikis, work packages, and repositories
in SQL for all projects and on-demand for a single project.

This removes all dependencies introduced from the ruby-based counting
(i.e., the acts_as_ountable gem).

myabc committed:
Merge pull request #3 from ulferts/fix/dev_to_rails4_branch_merge_specs

Fix/dev to rails4 branch merge specs

ulferts committed:
Merge pull request #3359 from floriank/feature/20248-error-collection

[20248] Display errors in the WP details pane via NotificationsService

ulferts committed:
hoping for spec stablility

myabc committed:
Revert "Temp. switch openproject-translations to PR branch"

This reverts commit 3fdfc147e8d115cf9365e33239fce2b5f1473958.

myabc committed:
Update openproject-translations

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix deprecated .update_all in MoveWorkPackageService

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix deprecation regressions in Repository specs

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Merge branch 'dev' into feature/merge-dev-into-rails4

Signed-off-by: Alex Coles <alex@alexbcoles.com>

NobodysNightmare committed:
Merge pull request #3362 from basiszwo/21176-space-character

[21176] Strip all passed values in settings area

oliverguenther committed:
Allow automatic creation of manageable repositories through a setting

This commit allows manageable repositories to be automatically created
by default whenever a new project is created and the repository module
is activated.

Relevant work package: https://community.openproject.org/work_packages/21277

oliverguenther committed:
Workaround to PG adapter returning non-integer aggregated values

oliverguenther committed:
Extend autofetching of commits into update repository information

This commit extends the previous functionality of *autofetch commits*
into internal updating of repository information, which encompasses:

* Commit information (changesets)
* Disk space information

They are now both retrieved when browsing the repository and the latter
is fetched asynchronously whenever the last update is older than the
timeout setting.

This timeout setting is now user-configurable.

basiszwo committed:
Fix seed data by adding missing attributes to work package

[ci skip]

basiszwo committed:
Extract development seed data to separate file

[ci skip]

NobodysNightmare committed:
Merge pull request #3363 from basiszwo/21279-fix-seed-data

[21279] Fix seed data for development environment

oliverguenther committed:
Refactor automatic repository setting into single select

This commit moves the automatic repository setting into a single
select and moves it below the enabled vendors.

Changing the enabled vendors will now also dynamically add or remove vendors
from the repository select field.
The select field is disabled when no option remains.

HDinger committed:
changed the z-index, so that the notifications are behind the menu

HDinger committed:
increased the z-index of the top menu to assure that it is higher and not equal to the notification box

oliverguenther committed:
Merge pull request #3373 from HDinger/fix/21289-notification-overlaps-menu

[21289] Notification box overlaps login menu

ulferts committed:
rewrite legacy spec as feature spec

Another option would have been to write it as unit tests for the
controller. However, that would have required to restructure the
controller as unit testing would otherwiese require to much mocking.

Additionally, the feature specs are required anyway.

oliverguenther committed:
Enable SCM config for spec

myabc committed:
Merge pull request #4 from ulferts/fix/time_entry_csv_export_spec

rewrite legacy spec as feature spec

ulferts committed:
Merge pull request #3339 from HDinger/fix/17579-long-subject-destroys-layout-details-pane

[17579] long subject destroys layout details pane

oliverguenther committed:
Better localized revision link using dynamically compiled date and href

ulferts committed:
Merge pull request #3327 from myabc/feature/merge-dev-into-rails4

[20040] Merge branch 'dev' into feature/rails4

ulferts committed:
Merge branch 'dev' into feature/rails4

oliverguenther committed:
Remove revision unit test example that requires compile

NobodysNightmare committed:
Merge pull request #3370 from timhabermaas/random-cleanups

Some smaller refactorings

ulferts committed:
Merge branch 'dev' into feature/rails4

ulferts committed:
Merge pull request #2104 from opf/feature/rails4

[20040] Upgrade to Rails 4.0.x

oliverguenther committed:
Use String concat instead of Array join

ulferts committed:
fix wrong method name broken when fixing merge conflict

oliverguenther committed:
Use present instead of presence

oliverguenther committed:
Remove PG workaround required for Rails < 4

NobodysNightmare committed:
remove workaround for type conversion in PSQL

as of Rails 4 this should no longer be neccessary

oliverguenther committed:
Test for Activity and Activity::Comment, which may also be initial

NobodysNightmare committed:
Merge pull request #3364 from basiszwo/git-cleanup-task

[ci skip] Add rake task to cleanup merged (remote) branches

oliverguenther committed:
Merge pull request #3372 from oliverguenther/feature/automatic_repository_creation

[21277] Allow automatic creation of manageable repositories through a setting

oliverguenther committed:
Merge pull request #3378 from NobodysNightmare/feature/remove_psql_type_workaround

Remove workaround for type conversion in PSQL

oliverguenther committed:
Merge pull request #3368 from oliverguenther/feature/projects_storage_refactor

[21242] Move project required disk space counting to SQL

NobodysNightmare committed:
Merge pull request #3358 from oliverguenther/feature/repository_view_hooks

Add better contextual view hooks to navigation and empty view

oliverguenther committed:
Merge pull request #3323 from oliverguenther/feature/repository_activity

[21047] Work package activities also display repository revisons

oliverguenther commented:
Superseded by https://github.com/opf/openproject/pull/3380
